spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/actions/BaseMigrateTableSparkAction.java

```diff
@@ -112,6 +112,8 @@ private MigrateTable.Result doExecute() {
     StagedSparkTable stagedTable = null;
     Table icebergTable;
     boolean threw = true;
+    boolean dropBackup =
+        Boolean.parseBoolean(additionalProperties().getOrDefault("dropBackup", "false"));
```

Comment:
So additionalProperties() looks like properties that will be set on the target table, right?  That's not great because then there will be this extra property 'dropBackup' that's useless for the Iceberg table.

So I think , let's add an extra method on MigrateTableSparkAction.dropBackup(boolean), and propagate it via its own dedicated argument in the procedure?

Reply:
I added the extra dropBackup method on MigrateTableSparkAction as you suggested in the latest commit. This way we avoid modifying the table properties.